climada/util/finance.py

```diff
@@ -181,6 +180,68 @@ def gdp(cntry_iso, ref_year, shp_file=None, per_capita=False):
     return close_year, close_val
 
 
+def download_world_bank_indicator(
+    country_code: str, indicator: str, parse_dates: bool = False
+):
+    """Download indicator data from the World Bank API for all years or dates on record
+
+    Parameters
+    ----------
+    country_code : str
+        The country code in ISO alpha 3
+    indicator : str
+        The ID of the indicator in the World Bank API
+    parse_dates : bool
+        Whether the dates of the indicator data should be parsed as datetime objects.
+        If ``False`` (default), they will be parsed as ``int`` (this only works for
+        yearly data).
+
+    Returns
+    -------
+    pd.Series
+        A series with the values of the indicator for all dates (years) on record
+    """
+    # Download data from API
+    raw_data = []
+    pages = np.inf
+    page = 1
+    while page <= pages:
+        response = requests.get(
+            f"https://api.worldbank.org/v2/countries/{country_code}/indicators/"
+            f"{indicator}?format=json&page={page}",
+            timeout=30,
+        )
+        json_data = json.loads(response.text)
+
+        # Check if we received an error message
+        try:
+            if json_data[0]["message"][0]["id"] == "120":
+                raise RuntimeError(
+                    "Error requesting data from the World Bank API. Did you use the "
+                    "correct country code and indicator ID?"
+                )
+        # If no, we should be fine
+        except KeyError:
+            pass
+
+        # Update the data
+        pages = json_data[0]["pages"]
+        page = page + 1
+        raw_data.append(json_data[1])
+
+    # Create dataframe
+    data = pd.concat([pd.DataFrame.from_records(rec) for rec in raw_data])
+
+    # Parse dates
+    if parse_dates:
```

Comment:
ideally the caller/user doesn't have to care whether or not the date has to be parsed to a `datetime` object.
I'd suggest to either derive `parse_dates` from the data itself or return it as is.

Reply:
The World Bank API data is most often stored in yearly intervals, but sometimes in other intervals, too (quarterly, monthly, etc.). We could first try to parse as `int`, and if that fails, try parsing as `DatetimeIndex` instead. However, I would not remove the parameter, because users may not want to retrieve DatetimeIndex even for integer data 🤔